ethereum/core/src/main/java/org/hyperledger/besu/ethereum/vm/operations/ModOperation.java

```diff
@@ -33,11 +35,14 @@ public Gas cost(final MessageFrame frame) {
 
   @Override
   public void execute(final MessageFrame frame) {
-    final UInt256 value0 = frame.popStackItem().asUInt256();
-    final UInt256 value1 = frame.popStackItem().asUInt256();
-
-    final UInt256 result = value0.mod(value1);
+    final UInt256 value0 = UInt256.fromBytes(frame.popStackItem());
+    final UInt256 value1 = UInt256.fromBytes(frame.popStackItem());
 
-    frame.pushStackItem(result.getBytes());
+    if (value1.isZero()) {
```

Comment:
This optimization is out of scope of this PR in my opinion.

Reply:
Same as https://github.com/hyperledger/besu/pull/215#discussion_r350645086

Reply:
perhaps a `mod0` could be added in Tuweni at a later date.

Reply:
Certainly.

Reply:
added and used in this PR.